<!doctype html>
<html lang="en-US">
<head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <meta name="author" content="Deren Eaton">
    <meta name="description" content="Evolutionary Biology">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>Fundamentals-4</title>
    <!-- CSS styling -->
    <link rel="stylesheet" href="../../reveal/css/reveal.css">
    <link rel="stylesheet" href="../../eaton-lab.css">
    <link rel="stylesheet" href="../../reveal/lib/css/zenburn.css">    
    <!-- <link rel="stylesheet" href="./custom.css"> -->
    <script>
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = window.location.search.match( /print-pdf/gi ) ? '../../reveal/css/print/pdf.css' : '../../reveal/css/print/paper.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>    
</head>

<!-- Not sure yet of this transition effect -->
<body style="transition: transform 0.8s ease 0s;"> 

    <!-- set reveal defaults ... -->
    <div class="reveal linear" data-transition-speed="default" data-background-transition="default">

        <!-- By giving slides a dimension we can use pixels or % -->
        <div class="slides" style="width:1600px; height:900px;">

            <!-- SLIDE 0 (TITLE) -->
            <section id="title" data-background="#262626" style="top:300.5px; display:block;" data-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 20%">
                <h1 class="title" style="color:#5D9B84">
                Fundamentals of Evolution</h1>
                <h5 style="color: lightgrey"> EEEB G6110</h5>
                <h4 style="color: white">Session 4: Intro to phylogenetics</h4>
                <br>
                </div>
            </section>

            <section id="contents" ddata-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 5%">
                <h2> Today's topics </h2>
                <br>
                <p style='font-size:1.25em; font-weight:400; color:#5D9B84; text-align:left; margin-left: 25%; margin-right:10%'>
                    1. Recap: Tree thinking
                    <br><br>
                    2. Intro to phylogenetics
                    <br><br>                    
                    3. Intro to model based inference
                    <br><br>
                </p>
                </div>
            </section>


            <section id="notebooks recap" class="future" data-transition="none">
                <div class="standard">
                <h3> Jupyter notebooks </h3>
                <br>
                <div class="textbox"><p>
                Web-based environment for creating and sharing reproducible code. Our example explored <i>tree-thinking</i> and <i>viewing trees as data</i> using the toytree package in <span style="font-weight:400">Python</span>.</p></div>
                <img src="../../data-img/toytree-example.png" style="width:70%">
                <br>

                <p style="font-size: 0.7em; margin-left: 15%; margin-right: 15%">
                <a href="https://toytree.readthedocs.io/">https://toytree.readthedocs.io/</a>
                </p>
                </div>
            </section>


            <section id="toytree 1" class="future" data-transition="none">
                <div class="standard">
                <h3> Tree thinking exercises </h3>
                <br>
                <div class="textbox"><p>
                Reading trees involves interpreting the order in which lineages
                share common ancestors by tracing relationships backwards from the tips towards the root. <i>Rotating nodes</i> does not affect these relationships, even though the order of the tips changes. 
                <span style="font-weight:400;color:#5D9B84">Which topology is different?</span>
                </p></div>
                <img src="../../data-img/toytree-thinking.png" style="width:70%">
                <br>
                </div>
            </section>


            <section id="toytree 2" class="future" data-transition="none">
                <div class="standard">
                <h3> Trees as data </h3>
                <br>
                <div class="textbox"><p>
                Phylogenetic trees are more than just pictures, they represent a data structure that can be interpreted and used in model-based analyses. Stored in <i>Newick</i> format.
                </p></div>
                <img src="../../data-img/toytree-newick-exercise-1.png" style="width:70%">
                <br>
                </div>
            </section>    


            <section id="toytree 3" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Trees as data</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                Phylogenetic trees are more than just pictures, they represent a data structure that can be interpreted and used in model-based analyses. Stored in <i>Newick</i> format. 
                </p>
                <img src="../../data-img/toytree-newick-exercise-2.png" style="width:70%">
                <br>
                </div>
            </section>


            <section id="quiz 1" class="future" data-transition="none">
                <div class='standard' style="margin-top:10%">
                <h2> Quiz time!</h2>
                </div>
            </section>


            <section id="quiz 2" class="future" data-transition="none" 
            data-background-iframe="https://www.polleverywhere.com/free_text_polls/AJurzwsFGpnm8NN3G0eha?preview=true"
            hidden="">
            </section>




            <section id="phylo inference why 1" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: why?</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                Much of evolutionary research involves reconstructing the past, or making inferences on the basis of relatedness/ancestry. Thus it is relevant to understand <span style="font-weight:400;">how</span> evolutionary relationships are inferred, and the level of 
                <span style="font-weight:400">confidence</span> we should place in various types of inference.
                </p>
                <img src="https://www.ebi.ac.uk/training/online/sites/ebi.ac.uk.training.online/files/resize/user/3638/documents/fig11-550x258.png" style="width:45%">
                <br>
                </div>
            </section>


            <section id="phylo inference why 2" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: why?</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                Dating evolutionary events: when populations or species diverged; whether different lineages show correlated histories; to compare rates of divergence among lineages; to compare histories of different genes; to infer ancestral states (geography, traits).
                </p>
                <img src="https://upload.wikimedia.org/wikipedia/commons/e/e4/ASR_phylogeny.png" style="width:45%">
                <br>
                </div>
            </section>


            <section id="phylo inference why 3" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: why?</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                Comparative methods: <i>species are non-independent data points</i>. Some share more evolutionary history than others, i.e., diverged from a common ancestor more recently. 
                <br><br>
                Example: Birds fly and lay eggs, mammals mostly do not fly or lay eggs. Is the correlation between flying and laying eggs an <i>adaptation</i>? Almost surely not, it is a coincident correlation due to the shared history of all mammals versus all birds, and the traits they inherited from their ancestors.
                </p>
                </div>
            </section>


            <section id="phylo example 1" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: examples</h3>
                <br>
                <div style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                <p> Methods of phylogenetic inference, and model-based historical inferences using trees, are both highly active areas of research. 
                Many new methods are published in the journal of 
                <a href="https://academic.oup.com/sysbio/issue">Systematic Biology</a>, while countless applied examples are published in various journals, including Evolution, Molecular Biology and Evolution, Molecular Phylogenetics and Evolution, Molecular Ecology, etc.</p>
                </div>
                </div>
            </section>


            <section id="sysbio" class="future" data-transition="none" 
            data-background-iframe="https://academic.oup.com/sysbio/issue"
            hidden="">
            </section>

            <section id="sysbio" class="future" data-transition="none" 
            data-background-iframe="https://www.polleverywhere.com/free_text_polls/JrlMeSWrI7LdDEviEnK5D?preview=true"
            hidden="">
            </section>


            <section id="phylo inf 1" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <div style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                <p> <span style="font-weight:400">Collect/measure homologous characters</span> for some number of taxa. For DNA, identifying homology typically involves targeting regions of the genome using primers, or mapping sequenced reads from the genome to the same region of a reference genome. Either way, it is based on <i>sequence similarity</i>. This is typically followed by a more rigorous <i>multiple sequence alignment</i>.
                </p>
                </div>
                <br>
                <img src="http://ugene.unipro.ru/wp-content/uploads/2015/03/55.png" style="width:30%">
                </div>
            </section>


            <section id="phylo inf 2" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <div style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                <p> 
                A general outline of phylogenetic inference:<br><br>
                    1. Propose a starting tree (e.g., random or star). <br>
                    2. Score based on some criterion (e.g., parsimony, likelihood, distance).<br>
                    3. Modify to propose a new tree, return to step 2.
                </p>
                </div>
                </div>
            </section>            


            <section id="phylo inf 3" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <div style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                <p> 
                <span style="font-weight:400">Distance methods:</span> <i>build</i> a tree based on pairwise distances (e.g., UPGMA and neighbor-joining).
                <br><br>

                <span style="font-weight:400">Parsimony methods:</span> <i>search</i> for best tree based on minimizing character changes along branches.
                <br><br>

                <span style="font-weight:400">Likelihood/statistical methods:</span> <i>search</i> for tree on which the observed data is most likely to have evolved under an assumed <i>model of evolution</i>.
                <br><br>

                <span style="font-weight:400">Other methods:</span> many are variants of likelihood statistical methods, but others exist as well, but the most popularly used are listed above.
                <br><br>
                </p>
                </div>
                </div>
            </section>            

            <!-- describe distance methods briefly -->
            <section id="distance 1" class="future" data-transition="none">
                <div class="standard">
                <h3> Phylogenetic inference: distance methods (UPGMA)</h3>
                <br>
                <div class="textbox" style="font-size:0.8em"><p> 
                1. A pairwise distance matrix is computed from characters.
                <img src="../../data-img/distance-tree-1.png" style="width:60%">
                <br>

                2. New internal node is created joining the shortest distance between points. Their branch lengths are set to half the distance, and a new matrix is computed to this node using average distance to its descendants. 
                <br><br>

                3. repeat from 1 until all internal nodes are added. <br>
                <img src="../../data-img/distance-tree-2.png" style="width:50%">
                <br><br>
                </p></div>
                <p style="font-size:0.7em">                
                <a href=https://en.wikipedia.org/wiki/UPGMA>https://en.wikipedia.org/wiki/UPGMA</a></p>
                </div>
            </section>         


            <section id="distance 2" class="future" data-transition="none">
                <div class="standard">
                <h3> Phylogenetic inference: distance methods (neighbor-joining)</h3>
                <br>
                <div class="textbox" style="font-size:0.8em"><p> 
                1. A pairwise distance matrix is computed from characters.
                <img src="../../data-img/distance-tree-1.png" style="width:60%">
                <br>

                2. (Starting from a star-tree initially), join two nodes with smallest distance to create an ancestor, transform distance matrix so that all pairwise distances are retained (uses entire matrix not just pairs).
                <br><br>

                3. repeat from 1 until all internal nodes are added. <br>
                <img src="../../data-img/distance-tree-2.png" style="width:50%">
                <br><br>
                </p></div>
                <p style="font-size:0.7em">                
                <a href=https://en.wikipedia.org/wiki/Neighbor_joining>https://en.wikipedia.org/wiki/Neighbor_joining</a></p>
                </div>
            </section>   


            <section id="distance 3" class="future" data-transition="none">
                <div class="standard">
                <h3> Phylogenetic inference: distance methods (summary)</h3>
                <br>
                <div class="textbox"><p> 
                <span style="font-weight:400">shortcomings:</span> They yield a single best tree but do not provide a score (e.g., likelihood) that could be used to compare against alternatives (how much better is it?)<br><br>

                Sensitive to model assumptions, such as molecular clock (UPGMA) or that distances are additive (i.e., when one gets longer another gets shorter; NJ).<br><br>

                <span style="font-weight:400">strengths:</span> it is very fast and computationally efficient. For this reason NJ has had a resurgence for use in population genomics where datasets ae very large and model assumptions are less likely to be violated among closely related populations (e.g., human populations).
                </p></div>
                </div>
            </section>            
            
            <!-- describe parsimony, why is this an improvement... -->
            <section id="parsimony 1" class="future" data-transition="none">
                <div class="standard">
                    <h3>Phylogenetic inference: Parsimony</h3>
                    <br>
                    <div class="textbox"><p>
                        A character matrix and a topology. Count the number of character state changes.
                    </p></div>
                    <br>
                </div>
                <img src="../../data-img/char-felsen-1.png">
                <img src="../../data-img/tree-felsen-1.png" 
                    style="width:30%;transform:rotate(-1.5deg);">
            </section>


            <!-- describe parsimony, why is this an improvement... -->
            <section id="parsimony 2" class="future" data-transition="none">
                <div class="standard">
                    <h3>Phylogenetic inference: Parsimony</h3>
                    <br>
                    <div class="textbox"><p>
                        A character matrix and a topology. Count the number of character state changes.
                    </p></div>
                    <br>
                </div>
                <img src="../../data-img/char-felsen-1.png" style="margin-bottom:5%">
                <img src="../../data-img/tree-felsen-2.png" 
                    style="width:20%;transform:rotate(180deg);margin-left:5%;">
            </section>            


            <!-- describe parsimony, why is this an improvement... -->
            <section id="parsimony 3" class="future" data-transition="none">
                <div class="standard">
                    <h3>Phylogenetic inference: Parsimony</h3>
                    <br>
                    <div class="textbox"><p>
                        A character matrix and a topology. Count the number of character state changes.
                    </p></div>
                    <br>
                </div>
                <img src="../../data-img/char-felsen-1.png" style="margin-bottom:14%">
                <img src="../../data-img/tree-felsen-3.png" 
                    style="width:20%;transform:rotate(-1.5deg);margin-left:5%;">
            </section>

            <!-- scores can be compared, distance trees cannot... -->
            <section id="parsimony 4" class="future" data-transition="none">
                <div class="standard">
                    <h3>Phylogenetic inference: Parsimony</h3>
                    <br>
                    <div class="textbox"><p>
                        <span style="font-weight:400">Parsimony principle:</span>
                        The evolutionary tree that minimizes the net amount of evolution (fewest steps) should be preferred. <br><br>

                        1. Propose a topology (e.g., randomly)<br>
                        2. Calculate score (e.g., Fitch algorithm)<br>
                        3. Propose new tree and repeat (test all or many trees).
                    </p></div>
                    <br>
                </div>
            </section>            


            <section id="parsimony 5" class="future" data-transition="none">
                <div class="standard">
                    <h3>Phylogenetic inference: Fitch algorithm</h3>
                    <br>
                    <div class="textbox" style="font-size:0.8em"><p>
                        We can count the number of steps required without needing to know the root of the tree, or which characters are ancestral versus derived. <br>

                        1. arbitrarily root the tree.<br>
                        <img src="../../data-img/tree-fitch-1.png" style="width:25%; margin-left:40%;"><br>
                        
                        2. bottom-up pass: Build <i>sets</i> for each internal node of the observed states. Union of states if they are different, intersection of states if they share a state.<br>
                        <img src="../../data-img/char-fitch-1.png"
                        style="width:20%">
                        <img src="../../data-img/tree-fitch-2.png" 
                        style="width:20%;margin-left:10%"><br>

                    </p></div>
                    <br>
                </div>
            </section>


            <section id="parsimony 6" class="future" data-transition="none">
                <div class="standard">
                    <div class="standard">
                    <h3>Phylogenetic inference: Fitch algorithm</h3>
                    <br>
                    <div class="textbox" style="font-size:0.8em"><p>
                        We can count the number of steps required without needing to know the root of the tree, or which characters are ancestral versus derived. <br><br>

                        3. top-down pass: Assign any nucleotide at the root set. To all children apply the same state as its parent if shared, else any other state in its set. Count transitions.<br>
                        <img src="../../data-img/tree-fitch-3.png" 
                        style="width:17%"><br>

                        <span style="font-weight:400">Is this better than an alternative tree?</span><br>
                        <img src="../../data-img/tree-fitch-4.png" 
                        style="width:17%"><br>
                    </p></div>
                    <br>
                </div>
            </section>


            <!-- but, tree search is a huuuuge problem... 
            spend some time here talking about what an unrooted versus rooted
            tree shows and why these numbers differ... 
            -->
            <section id="phylo inf 4" class="future" data-transition="none">
                <div class="standard">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <img src="../../data-img/inferring-trees-space.png">
                </div>
            </section>            


            <!-- parsimony has fallen out of use for DNA -->
            <section id="test" class="future" data-transition="none">
                <div class="standard">
                    <h3> Parsimony's pitfall </h3>
                    <br>
                    <div class="textbox">
                    <p> It does not account for homoplasy (repeated mutations to the same site). 
                    </div>
                    <img src="../../data-img/dist-homoplasy.png">
                </div>
            </section>

            <!-- homoplasy is widespread and must be modeled -->

            <!-- this is the goal of statistical approaches... next week.-->                      
            <section id="reading" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 5%">
                <h2> Reading for next session</h2>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">       
                <br>
                <a href="https://link.springer.com/article/10.1007/BF01734359">
                Felsenstein, Joseph. 1981. “Evolutionary Trees from DNA Sequences: A Maximum Likelihood Approach.” Journal of Molecular Evolution 17 (6): 368–76. https://doi.org/10.1007/BF01734359.</a>

                <br><br>
                <a href="https://www.annualreviews.org/doi/full/10.1146/annurev-ecolsys-110411-160331">
                O’Meara, Brian C. 2012. “Evolutionary Inferences from Phylogenies: A Review of Methods.” Annual Review of Ecology, Evolution, and Systematics 43 (1): 267–85. https://doi.org/10.1146/annurev-ecolsys-110411-160331.
                </a>
                <br><br>

                </div>
            </section>            




        <!-- SLIDES CLASS ENDING -->
        </div>

    <!-- REVEAL LINEAR CENTER CLASS ENDING -->
    </div>


    <!-- <div class="slide-number"></div> -->
    <!-- <div class="progress" style="display: block"><span style="width: 0px;"></span> -->
    <!-- JAVASCRIPT -->
    <script src="../../reveal/lib/js/head.min.js"></script>        
    <script src="../../reveal/js/reveal.js"></script>
    <script type="text/javascript" src="../../reveal/plugin/highlight/highlight.js"></script>
    <script>
        Reveal.initialize({
            controls:true, 
            progress:false, 
            loop:false, 
            history:true,
            center:false,
            fragmentInURL:true,
            margin:.05,
            minScale:0.1,
            maxScale:5,
            width:1600,
            height:900,
            theme:Reveal.getQueryHash().theme,
            transition:Reveal.getQueryHash().transition||"linear",
            slideNumber: true,
            // slideNumber: "h/v",
            dependencies: [
            //     { src: '../reveal/lib/js/classList.js', condition: function() { return !document.body.classList; } },
            { src: '../../reveal/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
            { src: '../../reveal/plugin/zoom-js/zoom.js', async: !0, condition: function() {return!!document.body.classList }},
            ]
        });

        // Reveal.configure({ slideNumber: true })
        // Reveal.configure({ slideNumber: "c/t" })            

        // Reveal.initialize({controls:!1,progress:!0,history:!0,center:!0,margin:.05,minScale:.1,maxScale:5,width:960,height:720,theme:Reveal.getQueryHash().theme,transition:Reveal.getQueryHash().transition||"linear",math:{mathjax:"http://cdn.mathjax.org/mathjax/latest/MathJax.js",config:"TeX-AMS_HTML-full"},dependencies:[{src:"../reveal/lib/js/classList.js",condition:function(){return!document.body.classList}},{src:"../reveal/plugin/markdown/marked.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/markdown/markdown.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/highlight/highlight.js",async:!0,callback:function(){hljs.initHighlightingOnLoad()}},{src:"../reveal/plugin/zoom-js/zoom.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/notes/notes.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/math/math.js",async:!0}]});
    </script>
</body>
</html>