---
layout: page
title: Welcome!
excerpt: "Welcome!"
modified: 2014-08-08T19:44:38.564948-04:00
image:
  feature:
  credit:
---

The Eaton lab is currently in the incipient stages of development with plans
to officially launch in the summer of 2017 in the [Department of Ecology,
Evolution, and Environmental Biology](http://e3b.columbia.edu)
at Columbia University. This site will serve as a venue for sharing updates
about the lab and its members, including notes on fieldwork, discussions of
papers, and posts about new methods being developed and used in the lab.
Check back soon for updates. Below you can find descriptions of several
major projects ongoing in the lab.

<br>
<h1 class="entry-subtitle" id="Flowers"
href="Flowers">Evolutionary consequences of interspecific reproductive interactions</h1>

The Hengduan Mountains of China on the eastern edge of the Tibetan plateau
is one of the most diverse temperate plant communities in the world.
Of the many clades that have diversified *in situ* in this region
*Pedicularis* stands out for its exceptional diversity
of floral forms displayed across >300 endemic species. A general link between
floral diversity and species diversity has long been recognized within flowering
plants, and the concentrated diversity of <em>Pedicularis</em> in the Hengduan
Mountains provides an ideal system in which to investigate the role that
<em>interspecific reproductive interactions</em> play in shaping this pattern.
The Eaton lab combines genomic analyses with field-based experiments and
community analyses to understand the role that gene flow and reproductive
interference play in shaping *Pedicularis* evolution.


<figure class="third">
	<a href="/images/pedics/P-p-batang.jpg">
		<img src="/images/pedics/P-p-batang.jpg" 
		alt="image"
		width='200px'>
	</a>
	<a href="/images/pedics/P-p-fetisowii.jpg">
		<img src="/images/pedics/P-p-fetisowii.jpg" 
		alt="image"
		width="200px">
	</a>  
	<a href="/images/pedics/P-p-lachnoglossa.jpg">
		<img src="/images/pedics/P-p-lachnoglossa.jpg"
		alt="image"
		width="200px">
	</a>    
  <figcaption>Floral diversity in <em>Pedicularis</em> of the Hengduan Mountains.</figcaption>
</figure>


<h1 class="entry-subtitle" id="Leaves"
href="Leaves">Replicated evolution of leaf form in <em>Viburnum</em></h1>

*Viburnum* is a widespread group of small trees and shrubs distributed primarily
throughout temperate regions of the northern hemisphere. The most rapid radiation
among extant *Viburnum* occurred in a clade that diversified throughout
neo-tropical cloud forests in the last 10My. In these narrow, isolated
environments stretching from Mexico to Bolivia, a distinct set of morpho-species
occurs repeatedly, representing distinct sets of co-varying leaf traits.
Phylogenomic analyses suggest that species within each geographical region are
each others closest relatives, but hybridization also makes their relationships
highly reticulate. We are combining genomic analyses with eco-physiological
studies to understand species relationships and the processes underlying this
apparent pattern of replicated evolution.

<figure class="third">
	<a href="/images/viburnies/acutifolium.jpg">
		<img src="/images/viburnies/acutifolium.jpg"
		alt="image"
		width="200px">
	</a>
	<a href="/images/viburnies/hybrid.jpg">
		<img src="/images/viburnies/hybrid.jpg"
		alt="image"
		width="200px">
	</a>
	<a href="/images/viburnies/sulcatum.jpg">
		<img src="/images/viburnies/sulcatum.jpg"
		alt="image"
		width="200px">
	</a>  
  <figcaption><em>Viburnum acutifolium</em>, a hybrid, and <em>V. sulcatum</em> from Oaxaca, Mexico.</figcaption>
</figure>


<h1 class="entry-subtitle" id="Phylo" href="Phylo">Computational phylogenetics and bioinformatics</h1>
I have been writing software primarily in the Python programming language
for a number of years now. This includes software for short-read (RAD-seq)
assembly (pyrad & ipyrad), for quartet species tree inference (tetrad), and
tree plotting (toytree). A major focus on my software development is
to create methods that make it easy to write and share code for reproducible
analyses, which when writing in Python has steered by development strongly
towards proselytizing for the use of jupyter-notebooks.  
