<!doctype html>
<html lang="en-US">
<head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <meta name="author" content="Deren Eaton">
    <meta name="description" content="Evolutionary Biology">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

    <title>Genomics-14</title>
    <!-- CSS styling -->
    <link rel="stylesheet" href="../../reveal/css/reveal.css">
    <link rel="stylesheet" href="../../eaton-lab.css">
    <link rel="stylesheet" href="../../reveal/lib/css/zenburn.css">    
    <!-- <link rel="stylesheet" href="./custom.css"> -->
    <script>
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = window.location.search.match( /print-pdf/gi ) ? '../../reveal/css/print/pdf.css' : '../../reveal/css/print/paper.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>    
</head>

<!-- Not sure yet of this transition effect -->
<body style="transition: transform 0.8s ease 0s;"> 

    <!-- set reveal defaults ... -->
    <div class="reveal linear" data-transition-speed="default" data-background-transition="default">

        <!-- By giving slides a dimension we can use pixels or % -->
        <div class="slides" style="width:1600px; height:900px;">

            <!-- SLIDE 0 (TITLE) -->
            <section id="title" data-background="#262626" style="top:300.5px; display:block;" data-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 20%">
                <h1 class="title" style="color:#5D9B84">
                Principles and Applications of Modern <br> DNA Sequencing</h1>
                <h5 style="color: lightgrey"> EEEB GU4055</h5>
                <h4 style="color: white">Session 14: Phylogenomics</h4>
                <br>
                </div>
            </section>


            <section id="contents" data-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 5%">
                <h2> Today's topics </h2>
                <br>
                <p style='font-size:1.2em; font-weight:400; color:#5D9B84; text-align:left; margin-left: 25%; margin-right:10%; line-height: 1.0'>
                <br>
                1. Phylogenomics introduction
                <br><br>
                2. The coalescent and why we do phylogenomics
                <br><br>
                3. Coalescent simulation (exercise)
                <br><br>
                4. Subsampling methods: anchored hybrid enrichment
                <br><br>
                5. Subsampling methods: RAD-seq (exercise)
                </p>
                </div>
            </section>



            <section class="future" data-transition="none">

                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Phylogenomic sampling </h3>
                    <br>
                    <div class="textbox"><p style='text-align:center'>
                    <!-- It's an exciting time to be asking these questions! Genomic technologies have <i>revolutionized</i> the study of ecology and evolution. -->
                    Characterize evolutionary history from <i>a subset</i> of sampled genomes (individuals). 
                    </p></div>
                    <br>
                    <img src="../../data-svg/intro-phylogenomic-sampling.svg" style="width:70%">
                    </div>
                </section>


                <section class="future" data-transition="none">
                    <img src="../../data-img/1KPG.png" style="width:45%">
                    <br>
                    <img src="../../data-img/1KPG-tree.png" style="width:45%">
                </section>

                <section class="future" data-transition="none">
                    <img src="../../data-svg/Spriggs.svg" style="width:70%">
                </section>
            </section>



            <section id="Coalescent" class="future" data-transition="none">

                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Phylogenomic sampling </h3>
                    <br>
                    <div class="textbox"><p style="text-align:center">
                         Characterize whole genomes from <i> a subset</i> of sequenced markers.
                    </p></div>   
                    <br>            
                    <img src="../../data-svg/intro-RAD-diagram.svg" style="width:70%">
                    </div>
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Genealogical variation </h3>
                    <br>
                    <div class="textbox"><p style="text-align:center">
                        It is important to examine evolutionary history across the entire genome. 
                    </p></div>
                    <img src="../../data-svg/intro-tree-outlines.svg" style="width:70%">
                    </div>
                </section>



                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Historical introgression/admixture </h3>                        
                    <br>
                    <div class="textbox"><p style="text-align:center">
                        It is important to examine evolutionary history across the entire genome. 
                    </p></div>
                    <img src="../../data-svg/intro-tree-outlines-2.svg" style="width:70%">
                    </div>
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> The Coalescent</h3>
                    <br>
                    <div class="textbox"><p>
                        A model that describes the expected waiting time until two or more
                        samples share a most recent common ancestor. The distribution of 
                        coalescent times within a population, or between populations, provides information about their history.
                        <br><br>
                        There are many genealogical histories that could possibly explain the genetic relatedness of a set of samples. We cannot observe the genalogies directly, only the sequence data that evolved on those genealogies.
                        <br><br>
                        Coalescent simulations provide a means to ask: "can the genetic variation that I observe in my samples be explained by neutral evolutionary processes?" 
                        </p></div>
                    </div>
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Population parameters (Ne)</h3>
                    <br>
                    <div class="textbox"><p>
                        The effective population size (Ne) of a population describes the probability that two samples share a common ancestor in the previous generation. This parameter does not translate directly to the actual population size, though they are likely correlated. Other factors like non-random mating and population structure also affect Ne. 
                    </p></div>
                    </div>
                </section>                



                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Single population model</h3>
                    <br>
                    <div class="textbox"><p>
                        If we assume that a population is randomly mating (panmictic) and neutrally evolving then the expected waiting time until n samples coalesce can be modeled entirely by Ne. 
                        <br><br>
                        Because n samples can share many possible genealogical histories (remember how big tree space is), and their genealogical relationships are expected to vary across their genomes (recombination makes different regions independent of others), we expect to observe a large variation in genealogical histories when examining many loci for n samples. 
                        <br><br>
                        <span style="font-weight:400">The coalescent model treats genealogies as a random varaible. We are interested in the expected <i>distribution</i> of variation when integrating over many genealogies.</span>
                    </p></div>
                    </div>
                </section>




                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Multiple population (structured) coalescent</h3>
                    <br>
                    <div class="textbox"><p>
                        When modeling multiple populations a "species tree" topology (e.g., "Species Tree") defines when different samples or their ancestors are able to share a parent in a previous generation. 
                        <br><br>
                        To predict the expected genetic similarity of samples in a structured coalescent model requires estimating Ne for each lineage as well as T, the divergence time of the populations. 
                        <br><br>
                        <span style="font-weight:400">Modern phylogenetic inference methods are based on the <i>multispecies coalescent model</i> which calculates the likelihood of observed genetic data given a set of parameters: Ne, T, and a topology. Searching over many topologies and many parameters can identify a best species tree model that explains variation among genealogies. 
                    </p></div>
                    </div>
                </section>



                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Coalescent Exercise</h3>
                    <br>
                    <a href=https://mybinder.org/v2/gh/genomics-course/2020-phylogenomics/master?filepath=nb-13.1-MSC.ipynb>Link to notebook 13.1 (MSC)</a>
                    </div>
                </section>
            </section>



            <section id="reading" class="future" data-transition="none">
                <section class="future" data-transition="none">
                    <img src="./data/Rokas-title.png">
                    <img src="./data/Rokas-trees.png">                    
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Rokas et al.: Discussion</h3>
                    <br>
                    <div class="textbox"><p style="text-align: center">
                        - What type of sequence data did they use?<br>
                        - Is this a shallow or deep phylogenetic question?<br>
                        - Why is there so much variation among gene trees?<br>
                        - What is their recommended solution (in 2002)?<br>
                        - Is this still a recommended method?<br>
                        <br><br>
                        - Would sampling more data help infer a better tree?
                    </p></div>
                    </div>
                </section>        


                <section class="future" data-transition="none">
                    <img src="./data/McCormack-title.png"><br>
                    <img src="./data/McCormack-tree.png" style="width:25%">
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> McCormack et al.: Discussion</h3>
                    <br>
                    <div class="textbox"><p style="text-align: center">
                        - What type of sequence data did they use?<br>
                        - Is this a shallow or deep phylogenetic question?<br>
                        - Is there agreement among the gene trees?<br>
                        - Is their species tree highly supported?<br>
                        <br><br>
                        - Would sampling more data help infer a better tree?
                    </p></div>
                    </div>
                </section>        
            </section>



            <section id="Inference methods" class="future" data-transition="none">
                <section class="future" data-transition="none">
                    <div style="margin-left: auto; margin-top: 5%">    
                    <h2> Phylogenomic inference methods</h2>
                    <img src="../../data-svg/consensus-pre.svg" style="width:55%">
                    </div>
                </section>

                <section class="future" data-transition="none">
                    <div style="margin-left: auto; margin-top: 5%">
                    <h2> Challenges: missing data</h2>
                    <img src="../../data-svg/consensus-pre2.svg" style="width:55%">
                    </div>
                </section>
            </section>


            <section id="datatypes" class="future" data-transition="none">
                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Preparing Genomic Libraries</h3>
                    <br>
                    <div class="textbox"><p>
                        Wet lab techniques for taking extracted DNA and ligating synethesized nucleotides to it to prepare it for a sequencing machine. <br><br>

                        Adapter sequences are oligonucleotides with a sequence that binds to some feature of the sequencing machine. <br><br>

                        Barcodes/Indices are unique molecular identifiers that can be ligated (attached) to DNA fragments so that they can be pooled for sequencing and later assigned to different samples based on the barcode (demultiplexed).
                    </p></div>
                    </div>
                </section>        



                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Targeted Hybrid Enrichment Methods</h3>
                    <br>
                    <div class="textbox"><p>
                        Methods for subsampling the genome to select particular regions for sequencing. Requires a priori knowledge about sequence at the regions of interest.
                        <br><br>
                        Design and order synethesized RNA baits that will bind to target DNA region. These baits are ligated to magnetic beads that allow them to be <i>pulled</i> out of solution with powerful magnets. This will <i>enrich</i> the DNA sample for the targeted regions.
                        <br><br>
                        Shotgun sequence the enriched library and assemble reads into contigs overlapping the targeted region.
                    </p></div>
                    </div>
                </section>        


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Exome sequencing (WES)</h3>
                    <br>
                    <div class="textbox"><p>
                        The exome is composed of all of the exons within the genome. It is different from the transcriptome, which contains all RNA transcribed in a cell. The transcriptome will vary among different cell types whereas the exome does not.<br><br>
                        Targeted exome sequencing uses hybrid target capture to enrich a DNA extraction for coding regions before shotgun sequencing. It requires a priori knowledge of the gene sequences. 
                        <br><br>
                        Whole Exome Sequencing is mostly used in human biomedical research, and model organism research, since designing an array or probe set for one species requires a high quality reference genome and is costly (i.e., needs to be used many times to recoup costs).
                    </p></div>
                    </div>
                </section>        


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Anchored hybrid enrichment methods</h3>
                    <br>
                    <div class="textbox"><p>
                        For phylogenomic analyses we typically do not need the whole exome, and instead design baits for just a subset of exons. In particular, exons that are highly conserved and occur as a single copy (not duplicated). 
                        RNA baits can be designed for many closely related taxa based on one or more closely related genomes. If the samples differ too much from the taxon used for bait design you end up with missing data.
                    </p></div>
                    </div>
                </section>        


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Ultraconserved Elements</h3>
                    <br>
                    <div class="textbox"><p>
                        Some genomic regions have been identified that are very very highly conserved among even very divergent taxa (e.g., all birds or all mammals). Sometimes these regions have unknown functions, some are related to important developmental genes. <br><br>

                        Baits have been designed that target these UCE regions and extend away from them for several hundred base pairs. The center has almost no variation but on the ends of contigs more variation is detected. 
                        <br><br>
                        Whereas it is often very hard to align orthologous regions among very distantly related species, UCEs seem to work well for obtaining many hundreds or thousands of orthologs.
                    </p></div>
                    </div>
                </section>        
            </section>




            <section id="RADseq" class="future" data-transition="none">

                <section class="future" data-transition="none">
                    <img src="./data/Andrews-title.png" style="width:40%">
                    <img src="./data/Andrews-fig.png" style="width:20%">
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> RAD-seq</h3>
                    <br>
                    <div class="textbox"><p style="text-align: center">
                        Subsample many thousands of regions across the genome without need to design baits. Fast and efficient subsampling method.
                        <br><br>
                        Initially used for association mapping, and genetic maps, where sparsely spaced markers are sufficient to identify ancestry relative to parents.
                        <br><br>
                        But because it is easy to generate thousands of markers it also became popular for population genetic and phylogenetic analyses. 
                    </p></div>
                    </div>
                </section>


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> Drawbacks of RAD-seq</h3>
                    <br>
                    <div class="textbox"><p style="text-align: center">
                        - Distantly related samples will not share the same restriction recognition sites (e.g., they accumulate mutations) and so it is characterized by a lot of missing data<br><br>
                        - <i>For organisms with small genomes </i> it is increasingly affordable for many types of questions to simply shotgun sequence the whole genome.
                    </p></div>
                    </div>
                </section>       


                <section class="future" data-transition="none">
                    <div class="standard">
                    <h3> In Silico Genomic Library Preparation Exercise</h3>
                    <br>
                    <a href=https://mybinder.org/v2/gh/genomics-course/2020-phylogenomics/master?filepath=nb-13.2-RAD-short.ipynb>Link to notebook 13.2 </a>
                    </div>
                </section>


            </section>





        <!-- SLIDES CLASS ENDING -->
        </div>

    <!-- REVEAL LINEAR CENTER CLASS ENDING -->
    </div>


    <!-- <div class="slide-number"></div> -->
    <!-- <div class="progress" style="display: block"><span style="width: 0px;"></span> -->
    <!-- JAVASCRIPT -->
    <script src="../../reveal/lib/js/head.min.js"></script>        
    <script src="../../reveal/js/reveal.js"></script>
    <script type="text/javascript" src="../../reveal/plugin/highlight/highlight.js"></script>
    <script>
        Reveal.initialize({
            controls:true, 
            progress:false, 
            loop:false, 
            history:true,
            center:false,
            fragmentInURL:true,
            margin:.05,
            minScale:0.1,
            maxScale:5,
            width:1600,
            height:900,
            theme:Reveal.getQueryHash().theme,
            transition:Reveal.getQueryHash().transition||"linear",
            slideNumber: true,
            // slideNumber: "h/v",

            dependencies: [
                //     { src: '../reveal/lib/js/classList.js', condition: function() { return !document.body.classList; } },
                { src: '../../reveal/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                // { src: '../../reveal/plugin/zoom-js/zoom.js', async: !0, condition: function() {return!!document.body.classList }},
                { src: '../../reveal/plugin/math/math.js', async: true }
            ]
        });

        // Reveal.configure({ slideNumber: true })
        // Reveal.configure({ slideNumber: "c/t" })            

        // Reveal.initialize({controls:!1,progress:!0,history:!0,center:!0,margin:.05,minScale:.1,maxScale:5,width:960,height:720,theme:Reveal.getQueryHash().theme,transition:Reveal.getQueryHash().transition||"linear",math:{mathjax:"http://cdn.mathjax.org/mathjax/latest/MathJax.js",config:"TeX-AMS_HTML-full"},dependencies:[{src:"../reveal/lib/js/classList.js",condition:function(){return!document.body.classList}},{src:"../reveal/plugin/markdown/marked.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/markdown/markdown.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/highlight/highlight.js",async:!0,callback:function(){hljs.initHighlightingOnLoad()}},{src:"../reveal/plugin/zoom-js/zoom.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/notes/notes.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/math/math.js",async:!0}]});
    </script>
</body>
</html>