<!doctype html>
<html lang="en-US">
<head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <meta name="author" content="Deren Eaton">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>Session-5</title>
    <link rel="stylesheet" href="../../reveal/css/reveal.css">
    <link rel="stylesheet" href="../../eaton-lab.css">
    <link rel="stylesheet" href="../../reveal/lib/css/zenburn.css">    
</head>

<!-- Not sure yet of this transition effect -->
<body style="transition: transform 0.8s ease 0s;"> 

    <!-- set reveal defaults ... -->
    <div class="reveal linear" data-transition-speed="default" data-background-transition="default">

        <!-- By giving slides a dimension we can use pixels or % -->
        <div class="slides" style="width: 960px; height: 720px; transform: translate(-50%, -50%) scale(0.969792) translate(50%, 50%);">

<!-- ###################################################################### -->


<!-- SLIDE 0 (TITLE) -->
<section 
	data-background="#262626" 
 	style="top:300.5px; display:block;"
 	class="present">
    <div style="margin-left: auto; margin-top: 20%">
    <h2 class="title">Principles and Applications of Modern DNA Sequencing</h2>
    <p style="color: lightgrey"> EEEB GU4055</p>
    <h5 style="color: white">Session 5: Phylogenetics I</h5>
    <br>
    </div>
</section>

<!-- SLIDE 1 (what are we doing today?)-->
<section
    data-background="white" 
    data-transition="none" 
    class="future"
    style="top: -20px; display: none;" 
    hidden="">
    <h3> Today's topics </h3>

    <br>
    <p style='font-size:0.9em; 
    text-align: left; margin-left: 100px; margin-right: 100px'>
        <br>
        1. Review notebook assignments: BLAST and Homology.
        <br><br>
        2. Discuss the assigned reading: OrthoDB.
        <br><br>
        3. Introduce new topic: Phylogenetics.
        <br><br>
    </p>
</section>


<!-- SLIDE 3 (notebook 1.0 review)-->
<section
    data-background="white" 
    data-transition="none" 
    class="future"
    style="top: -20px; display: none;" 
    hidden="">

    <section>
        <h3> Notebook 4.1: Homology and BLAST</h3>
        <p style="font-size:0.75em"> 
    What do we mean by orthology and paralogy?
        </p>
    </section>

    <section>
        <h3> Notebook 4.1: Homology and BLAST</h3>
        <p style="font-size:0.75em"> 
    What do we mean by orthology and paralogy?
        </p>
    <img src="http://ecoevo.unit.oist.jp/lab/wp-content/uploads/2013/08/GeneHomology_II.png" style="width: 80%; height: 25%;">    
    </section>

    <section>
        <h3> Notebook 4.1: Homology and BLAST</h3>
        <p style="font-size:0.75em"> 
    What do we mean by orthology and paralogy? Hemoglobin example.
        </p>
    <img src="https://schaechter.asmblog.org/.a/6a00d8341c5e1453ef01a73dde119f970d-800wi" style="width: 60%; height: 25%;">    
    </section>


    <section>
        <h3> Notebook 4.1: Homology and BLAST</h3>
        <p style="font-size:0.72em;"> 
    Question [1]: Using orthology to transfer the gene functional annotation from one organism to another might be inaccurate in some instances. What type of things do you think could lead to these inaccuracies? In other words, why is the function of the model organism gene not always the same as some other organism being compared to it? 
    </section>
</section>


<section 
    data-transition="none"
    data-background-iframe="https://blast.ncbi.nlm.nih.gov/Blast.cgi"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    



<section 
    data-transition="none"
    data-background-iframe="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&PAGE_TYPE=BlastSearch&LINK_LOC=blasthome"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    




<section 
    data-transition="none"
    data-background-iframe="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Get&RID=5P0R1P6E016"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    




<section 
    data-transition="none"
    data-background-iframe="https://www.ncbi.nlm.nih.gov/nuccore/?term=foxp2+%5BGENE%5D+AND+%22exon+7%22"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    




<section 
    data-transition="none"
    data-background-iframe="https://www.orthodb.org/"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    


<section>
    <h3> Notebook 4.1: OrthoDB</h3>
<img src="./data/orthodb.jpg" style="width: 70%; height: 25%;">    
</section>


<section 
    data-transition="none"
    data-background-iframe="
    https://www.orthodb.org/?level=2759&species=314295&query=FOXP2"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>



<section 
    data-transition="none"
    data-background-iframe="https://www.orthodb.org/?level=314295&species=314295&query=FOXP2"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    



<section>

    <section>
    <h3> Notebook 4.2: BLAST, NCBI, and APIs</h3>
    <p style="font-size:0.75em"> 
    BLAST is an algorithm for comparing sequences, it searches the NCBI 
    database. The NCBI Entrez tools provide an API interface to this database.
    These tools have facilitated the development of complex tools like 
    OrthoDB. 
    </p>       
    <img src="https://www.researchgate.net/publication/329761758/figure/fig2/AS:705385665159170@1545188456706/SCKEs-system-architecture-SCKE-is-a-Python-application-run-on-an-Apache-web-server.png" style="width: 65%; height: 25%;">    
    </section>

    <section>
        <h3> Notebook 4.2: Requests to query APIs</h3>      
        <pre ><code class="python" style="max-height: 500px">
  import requests

  # search term
  term = "FOXP2[GENE] AND Mammalia[ORGN] AND phylogenetic study[PROP]"

  # make a request to esearch 
  res = requests.get(
      url="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi", 
      params={
          "db": "nucleotide",
          "term": term,
          "sort": "Organism Name",
          "retmode": "text",
          "retmax": "20",
          "tool": "genomics-course", 
          "email": "de2356@columbia.edu",
          },
      )
        </pre></code>
    </section>

    <section>
              <h3> Notebook 4.2: Requests to query APIs</h3>      
        <pre ><code class="python" style="max-height: 500px">
  # returns a response object
  res = requests.get(...)

  # the URL is built from the params arguments to requests.get()
  res.url
        </pre></code>
        <pre><code class="python" style="background: grey">
  'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=nucleotide&term=FOXP2%5BGENE%5D+AND+Mammalia%5BORGN%5D+AND+phylogenetic+study%5BPROP%5D&sort=Organism+Name&retmode=text&retmax=20&tool=genomics-course&email=de2356%40columbia.edu'
        </pre></code>          
    </section>


    <section>
              <h3> Notebook 4.2: Requests to query APIs</h3>      
        <pre ><code class="python" style="max-height: 500px">
  # returns a response object
  res = requests.get(...)

  # the text attribute store the returned HTML
  res.text
        </pre></code>
        <pre><code class="python" style="background: grey">
<!-- <?xml version="1.0" encoding="UTF-8" ?>\n<!DOCTYPE eSearchResult PUBLIC "-//NLM//DTD esearch 20060628//EN" "https://eutils.ncbi.nlm.nih.gov/eutils/dtd/20060628/esearch.dtd">\n<eSearchResult><Count>62</Count><RetMax>20</RetMax><RetStart>0</RetStart><IdList>\n<Id>22596910</Id>\n<Id>22596958</Id>\n<Id>24496240</Id>\n<Id>263198301</Id>\n<Id>263198296</Id>\n<Id>22596906</Id>\n<Id>22596954</Id>\n<Id>22596956</Id>\n<Id>24496234</Id>\n<Id>39762715</Id>\n<Id>263198298</Id>\n<Id>39762716</Id>\n<Id>39762714</Id>\n<Id>263198295</Id>\n<Id>24496236</Id>\n<Id>263198299</Id>\n<Id>263198300</Id>\n<Id>22596912</Id>\n<Id>263198302</Id>\n<Id>512739739</Id>\n</IdList><TranslationSet><Translation>     <From>Mammalia[ORGN]</From>     <To>"Mammalia"[Organism]</To>    </Translation></TranslationSet><TranslationStack>   <TermSet>    <Term>FOXP2[GENE]</Term>    <Field>GENE</Field>    <Count>1853</Count>    <Explode>N</Explode>   </TermSet>   <TermSet>    <Term>"Mammalia"[Organism]</Term>    <Field>Organism</Field>    <Count>72234311</Count>    <Explode>Y</Explode>   </TermSet>   <OP>AND</OP>   <TermSet>    <Term>phylogenetic study[PROP]</Term>    <Field>PROP</Field>    <Count>6046242</Count>    <Explode>N</Explode>   </TermSet>   <OP>AND</OP>  </TranslationStack><QueryTranslation>FOXP2[GENE] AND "Mammalia"[Organism] AND phylogenetic study[PROP]</QueryTranslation></eSearchResult>\n' --> 
        </pre></code>          
    </section>    
</section>


<section 
    data-transition="none"
    data-background-iframe="https://www.ncbi.nlm.nih.gov/genbank/"
    style="top:-960px; display: block;" 
    class="future" 
    hidden="">       
</section>    



<section>
  <section>
    <h3> Phylogenetics I</h3>
    <p style="font-size:0.75em"> 
    A practical guide to inferring phylogenetic trees.
    </p>       
    <img src="https://www.digitalatlasofancientlife.org/wp-content/uploads/2017/06/PhylogeneticTree-Parts.png" style="width: 65%; height: 25%;">    
  </section>
</section>



</div>
</div>
    
    <!-- <div class="slide-number"></div> -->
    <!-- <div class="progress" style="display: block"><span style="width: 0px;"></span> -->


        <!-- JAVASCRIPT -->
        <script src="../../reveal/lib/js/head.min.js"></script>        
        <script src="../../reveal/js/reveal.js"></script>
        <script type="text/javascript" src="../../reveal/plugin/highlight/highlight.js"></script>
        <script>
            Reveal.initialize({
                controls:true, 
                progress:false, 
                loop:true, 
                history:true,
                center:false,
                fragmentInURL:true,
                margin:.05,
                minScale:0.1,
                maxScale:5,
                width:960,
                height:720,
                theme:Reveal.getQueryHash().theme,
                transition:Reveal.getQueryHash().transition||"linear",
                // slideNumber: true,
                // slideNumber: "h/v",
                dependencies: [
                //     { src: '../reveal/lib/js/classList.js', condition: function() { return !document.body.classList; } },
                { src: '../../reveal/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                ]
            });

            // Reveal.configure({ slideNumber: true })
            // Reveal.configure({ slideNumber: "c/t" })            

            // Reveal.initialize({controls:!1,progress:!0,history:!0,center:!0,margin:.05,minScale:.1,maxScale:5,width:960,height:720,theme:Reveal.getQueryHash().theme,transition:Reveal.getQueryHash().transition||"linear",math:{mathjax:"http://cdn.mathjax.org/mathjax/latest/MathJax.js",config:"TeX-AMS_HTML-full"},dependencies:[{src:"../reveal/lib/js/classList.js",condition:function(){return!document.body.classList}},{src:"../reveal/plugin/markdown/marked.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/markdown/markdown.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/highlight/highlight.js",async:!0,callback:function(){hljs.initHighlightingOnLoad()}},{src:"../reveal/plugin/zoom-js/zoom.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/notes/notes.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/math/math.js",async:!0}]});
        </script>
    </body>
</html>