<!doctype html>
<html lang="en-US">
<head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <meta name="author" content="Deren Eaton">
    <meta name="description" content="Evolutionary Biology">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <title>Fundamentals-8</title>
    <!-- CSS styling -->
    <link rel="stylesheet" href="../../reveal/css/reveal.css">
    <link rel="stylesheet" href="../../eaton-lab.css">
    <link rel="stylesheet" href="../../reveal/lib/css/zenburn.css">    
    <!-- <link rel="stylesheet" href="./custom.css"> -->
    <script>
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = window.location.search.match( /print-pdf/gi ) ? '../../reveal/css/print/pdf.css' : '../../reveal/css/print/paper.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>    
</head>

<!-- Not sure yet of this transition effect -->
<body style="transition: transform 0.8s ease 0s;"> 

    <!-- set reveal defaults ... -->
    <div class="reveal linear" data-transition-speed="default" data-background-transition="default">

        <!-- By giving slides a dimension we can use pixels or % -->
        <div class="slides" style="width:1600px; height:900px;">

            <!-- SLIDE 0 (TITLE) -->
            <section id="title" data-background="#262626" style="top:300.5px; display:block;" data-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 20%">
                <h1 class="title" style="color:#5D9B84">
                Fundamentals of Evolution</h1>
                <h5 style="color: lightgrey"> EEEB G6110</h5>
                <h4 style="color: white">Session 10: Phylogenetics II</h4>
                <br>
                </div>
            </section>

            <section id="contents" data-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 5%">
                <h2> Recap topics </h2>
                <br>
                <p style='font-size:1.2em; font-weight:400; color:#5D9B84; text-align:left; margin-left: 25%; margin-right:10%; line-height: 1.0'>
                    1. Homology 
                    <br><br>
                    2. Genealogy, gene trees and species trees
                    <br><br>
                    3. Trees as data
                    <br><br>
                    4. Phylogenetic inference
                </p>
                </div>
            </section>


            <section id="contents" data-transition="none" class="present">
                <div style="margin-left: auto; margin-top: 5%">
                <h2> Section topics </h2>
                <br>
                <p style='font-size:1.2em; font-weight:400; color:#5D9B84; text-align:left; margin-left: 25%; margin-right:10%; line-height: 1.0'>
                    1. Phylogenetic inference
                    <br><br>
                    2. Likelihood and model-based inference
                </p>
                </div>
            </section>            


            <section class="future" data-transition="none">
                <div class="standard">
                    <h3>Descent and common ancestry </h3>
                    <img src=./data/baum-genealogy-1.png style="width:30%">
                    <img src=./data/baum-genealogy-2.png style="width:30%">                    
                </div>
            </section>



           <section class="future" data-transition="none">
                <div class="standard">
                    <h3>Homology, homoplasy and Hemiplasy </h3>
                    <div class="textbox"><p>
                        Are the characters descended from a common ancestor? Has the character 
                        state evolved multiple times? Does the evolutionary relationships of 
                        genes controlling this trait match the evolutionary relationships of 
                        the species?
                    </p>
                        <img src=../8/sptree-2.png style="width:42%">                 
                    </div>
                </div>
            </section>


            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 5%">    
                <h2> Coding characters</h2>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%"> 
                <span style="font-weight: 400">Discrete data:</span> DNA is discrete, there are only four possible states (A,C,G,T). But what about body mass, height, color, etc.<br><br>

                <span style="font-weight: 400">Characters weighting:</span> Is one trait more informative than another? How to avoid biased sampling of characters?<br><br>

                <span style="font-weight: 400">Why not only use DNA?:</span> For phylogenetics of extant taxa we mostly do. But morphology is still very relevant to phylogeny of extinct taxa (e.g., dinosaurs) and placing fossils with extant taxa
                </p>
                </div>
            </section>


            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 5%">    
                <h2> DNA (nucleotides) as characters</h2>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%"> 
                <span style="font-weight: 400">heritable:</span> DNA and amino acid differences are heritable, not plastic.<br><br>

                <span style="font-weight: 400">homology:</span> can be inferred by comparing sequences using alignment algorithms. The spatial location of nucleotides in the genome provides evidence of its homology. <br><br>

                <span style="font-weight: 400">models:</span> informed evolutionary models can be used to estimate rates of character change (e.g., random or non-random rates).<br><br>

                <span style="font-weight: 400">availability and bias:</span> Huge amounts of data are available and there is no inherent bias or expectation that certain parts of the genome (characters) should support one hypothesis more than another.
                </p>
                </div>
            </section>



            <section id="phylo inf 1" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <div style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                <p> <span style="font-weight:400">Collect/measure homologous characters</span> for some number of taxa. For DNA, identifying homology typically involves targeting regions of the genome using primers, or mapping sequenced reads from the genome to the same region of a reference genome. Either way, it is based on <i>sequence similarity</i>. This is typically followed by a more rigorous <i>multiple sequence alignment</i>.
                </p>
                </div>
                <br>
                <img src="http://ugene.unipro.ru/wp-content/uploads/2015/03/55.png" style="width:30%">
                </div>
            </section>


            <section id="phylo inf 2" class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <div style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                <p> 
                A general outline of phylogenetic inference:<br><br>
                    1. Propose a starting tree (e.g., random or star). <br>
                    2. Score based on some criterion (e.g., distance, parsimony, likelihood).<br>
                    3. Modify to propose a new tree, return to step 2.
                </p>
                </div>
                </div>
            </section>



            <!-- describe parsimony, why is this an improvement... -->
            <section id="parsimony 3" class="future" data-transition="none">
                <div class="standard">
                    <h3>Phylogenetic inference: Parsimony</h3>
                    <br>
                    <div class="textbox"><p>
                        A character matrix and a topology. Count the number of character state changes.
                    </p></div>
                    <br>
                </div>
                <img src="../../data-img/char-felsen-1.png" style="margin-bottom:14%">
                <img src="../../data-img/tree-felsen-3.png" 
                    style="width:20%;transform:rotate(-1.5deg);margin-left:5%;">
            </section>



            <!-- but, tree search is a huuuuge problem... 
            spend some time here talking about what an unrooted versus rooted
            tree shows and why these numbers differ... 
            -->
            <section id="phylo inf 4" class="future" data-transition="none">
                <div class="standard">
                <h3> Phylogenetic inference: methods</h3>
                <br>
                <img src="../../data-img/inferring-trees-space.png">
                </div>
            </section>            


            <!-- parsimony has fallen out of use for DNA -->
            <section id="test" class="future" data-transition="none">
                <div class="standard">
                    <h3> Parsimony's pitfall: Homoplasy </h3>
                    <div class="textbox">
                    <p> It does not account for homoplasy (repeated mutations to the same site). 
                        Or other patterns of varying rates of character change.
                    </p>
                    </div>
                    <img src="../../data-img/dist-homoplasy.png">
                </div>
            </section>


            <section class="future" data-transition="none">
                <div class="standard">
                <h3> Inferring the past from the present </h3>
                <br>
                <div class="textbox"><p>
                Statistical modeling is widely used in evolutionary research
                to test or compare hypotheses and to estimate model parameters.
                <br><br>
                A model describes a mechanistic (probabilistic) process that could produce observable data. It is especially useful when  
                <span style="font-style:italic;font-weight:400">we cannot observe the past, but we can observe data at the present.</span>
                </div>
            </section>    



            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Models, Probability, and Likelihood</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">
                The use of statistical models for <i>historical inference</i> 
                is not unique to evolution, or biology, although many statistical methods were developed by biologists <a href="https://en.wikipedia.org/wiki/Likelihood_function">(e.g., the Likelihood framework was developed by the geneticist R.A. Fisher)</a>
                <br><br>

                A <i>likelihood function</i> describes the probabilitiy of a set of <i>observations</i> given a set of model <i>parameters</i>. 
                Likelihood can express the probability of a DNA sequence alignment given a phylogenetic hypotheses, or it can express the probability that a coin toss will turn up heads versus tails. 
                <br><br>
                What do we mean by a model parameter?
                </p>
                </div>
            </section>



            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Models, Probability, and Likelihood</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">

                <span style="font-weight:400">The coin toss example:</span>
                We can describe a statistical model for the outcome of a coin toss. For a single trial (a toss) there are only two possible outcomes, H or T. The probability of H is unknown. If the coin is fair then p=0.5, but maybe p!=0.5. <br><br>

                The probability of observing data ($D$) given model parameters ($\Theta$) is described by a <i>likelihood function</i>: <br>
                
                $$ L(\Theta) = f(D | \Theta)$$
                <br>

                The observed data are fixed. The function evaluates the data given proposed values for $\Theta$ and returns a score (the likelihood) for how well the parameters explain the observed data under a given model.
                The goal of <span style="font-weight:400">maximum likelihood</span>
                is to estimate $\Theta$ by finding the value that maximizes $L(\Theta)$
                </p>
                </div>
            </section>            



            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                <h3> Models, Probability, and Likelihood</h3>
                <br>
                <p style="font-size: 0.9em; text-align: left; margin-left: 15%; margin-right: 15%">

                <span style="font-weight:400">The coin toss example:</span>
                
                Imagine we observe the following sequence of tosses:
                <br><br>

                $$ H, T, T, H, H, H, T, T, H, H, T, T, T$$
                <br>

                There are multiple ways in which to describe a likelihood function based on this observation to test different hypotheses. One  hypothesis we may wish to ask is whether the probability of <i>each trial</i> is fair (prob. heads = 0.5).
                <br><br>
                By <a href=https://en.wikipedia.org/wiki/Probability_theory>probability theory</a>, the outcome of many independent events is the product of their probabilities. For example, the probability of flipping heads 5 times in a row using a fair coin is: 
                <br><br>
                $$ 0.5 * 0.5 * 0.5 * 0.5 * 0.5 = 0.03125 $$
                </p>
                </div>
            </section>


            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                    <h3> Models, Probability, and Likelihood</h3>
                    <br>
                    <div class="textbox">
                        <p>
                        <span style="font-weight:400">The coin toss example:</span>
                
                        In other words, <a href=https://en.wikipedia.org/wiki/Joint_probability_distribution>the joint probability</a> of multiple independent events is computed as:<br><br>

                        $$ P(A, B) = P(A) * P(B) $$
                        <br>
                        In other words, the joint probability can be calculated as the product of many <i>independent</i>probabilities:<br><br>
                        $$ P(A_{i}) = \Pi_{i} P(A_{i}) $$
                        <br>
                        The likelihood of 5 heads in a row when p=0.5 is very low.
                        <br><br>
                        $$ L(\Theta=0.5) = 0.5 * 0.5 * 0.5 * 0.5 * 0.5 = 0.03125 $$
                        </p>
                    </div>
                </div>
            </section>  


            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                    <h3> Models, Probability, and Likelihood</h3>
                    <br>
                    <div class="textbox">
                        <p>
                        <span style="font-weight:400">The coin toss example:</span>
                
                        In other words, <a href=https://en.wikipedia.org/wiki/Joint_probability_distribution>the joint probability</a> of multiple independent events is computed as:<br><br>

                        $$ P(A, B) = P(A) * P(B) $$
                        <br>
                        In other words, the joint probability can be calculated as the product of many <i>independent</i>probabilities:<br><br>

                        $$ P(A_{i}) = \Pi_{i} P(A_{i}) $$
                        <br>
                        <i><span style="font-weight:400;">But, the likelihood of 5 heads in a row when p=0.99 is quite high. </span></i>
                        <br><br>
                        $$ L(\Theta=0.99) = 0.99 * 0.99 * 0.99 * 0.99 * 0.99 = 0.9509 $$
                        </p>
                    </div>
                </div>
            </section>   




            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                    <h3> The likelihood function</h3>
                    <br>
                    <div class="textbox">
                        <p>
                        <span style="font-weight:400">The coin toss example:</span>
                
                        The likelihood function for the probability of heads based on observed coin tosses describes the probability that a coin comes up heads (p) or tails (1-p). Since tails is just one minus heads there is only one parameter (p). The observed data is simply the number of trials (n) and the number of trials that came up heads (h):
                        <br><br>
                        $$ L(p) = p^{h} * (1-p)^{n-h}$$
                        <br>
                        <i><span style="font-weight:400;">The likelihood score depends only on the parameter (p) and the observation (n, h). </span></i>
                        <br><br>
                        $$ L(\Theta=0.5) = (0.5)^{5} * (1-0.5)^{0}$$
                        <br>              
                        $$ L(\Theta=0.99) = (0.99)^{5} * (1-0.99)^{0} $$
                        </p>
                    </div>
                </div>
            </section>                                    


            <section class="future" data-transition="none">
                <div style="margin-left: auto; margin-top: 2%">
                    <h3> Exercise in Notebook</h3>
                    <br>
                    <div class="textbox"><p>
                        <a href=https://mybinder.org/v2/gh/genomics-course/f5-likelihood.git/master?filepath=notebooks>https://mybinder.org/v2/gh/genomics-course/f5-likelihood.git/master?filepath=notebooks</a>
                    </p></div>
                </div>
            </section>            






            <section class="future" data-transition="none">
                <div class="standard">
                <h3> Statistical phylogenetics</h3>
                <br>
                    <div class="textbox">
                    <p>
                    The likelihood function calculates probability of observing a character state (DNA site in an alignment) under a specific model of molecular substitution and a phylogenetic tree (more on these soon).<br><br>

                    Simplest models treat each DNA site independently so that the likelihood of a sequence alignment (many sites) is simply the product of the individual site likelihoods: <br><br>

                    $$ L(D) = \prod_{i=1}^{n} L(D_{i}) = \prod_{i=1}^{n} f(D_{i} | \Theta) $$
                    </p>
                    </div>
                </div>
            </section>



            <section class="future" data-transition="none">
                <div class="standard">
                <h3> Statistical phylogenetics</h3>
                <br>
                    <div class="textbox">
                    <p>
                    By <span style="font-weight:400">only knowing the results of coin tosses that occurred in the past</span> we are able to estimate the <i>model parameters that are most likely to have produced them</i> (i.e., the probability the coin toss is heads or tails).<br><br>

                    The same principle applies to estimating the evolutionary distance between two aligned DNA sequences. Looking <span style="font-weight:400">only at the results of an evolutionary process that occurred in the past</span> we can estimate the model parameters that are most likely to have produced the DNA differences between a set of taxa. 
                    <br><br>

                    Whereas the coin toss problem assumed a <i>Bernoulli distribution</i> as the underlying model, for DNA substitutions we use a molecular substitution model, which is a type of <span style="font-weight:400">Continuous Time Markov Model</span>.
                    </p>
                    </div>
                </div>
            </section>

            <!-- parsimony has fallen out of use for DNA -->
            <section class="future" data-transition="none">
                <div class="standard">
                    <h3> Molecular Substitution Models</h3>
                    <br>
                        <div class="textbox">
                        <p> Many mutations occur that are not observable in the present-daty samples alone due to <i>homoplasy</i> (repeated mutations to the same sites). 
                        </p></div>
                    <img src="../../data-img/dist-homoplasy.png">
                </div>
            </section>



            <!-- parsimony has fallen out of use for DNA -->
            <section class="future" data-transition="none">
                <div class="standard">
                    <h3> Molecular Substitution Models</h3>
                    <br>
                    <div class="textbox">
                    <p> 
                    <i>Instantaneous transition rate matrix ($Q$): </i>
                    Similar to the coin toss example, where q was equal to 1-p, you can see that the probability of not changing over some length of time is simply (1 - probability of changing.)
                    </p></div>

                    <img src="https://www.oreilly.com/library/view/basic-applied-bioinformatics/9781119244332/images/bapp06f02.gif" style="width:40%">

                    <br><br><br>
                    <a href=https://en.wikipedia.org/wiki/Models_of_DNA_evolution>https://en.wikipedia.org/wiki/Models_of_DNA_evolution</a>
                </div>
            </section>


            <section class="future" data-transition="none">
                <div class="standard">
                    <h3> Molecular Substitution Models</h3>
                    <br>
                    <div class="textbox">
                    <p> 
                    <i>Transition-probability matrix:</i> the probability of change between states over some time interval (t) is easily calculated as $ P(t) = e^{Qt} $. This tells us the probability of starting in some state (e.g., A) and ending as another (e.g., T). Jukes-Cantor Model predicts equal prob. for any state over very long branches.
                    </p></div>
                    <img src="https://upload.wikimedia.org/wikipedia/commons/4/4f/Jukes_and_Cantor_%281969%29_JC69_Pij_Pii.jpg" style="width:25%">
                    <br>
                    <a href=https://en.wikipedia.org/wiki/Models_of_DNA_evolution>https://en.wikipedia.org/wiki/Models_of_DNA_evolution</a>
                </div>
            </section>            


            <section class="future" data-transition="none">
                <div class="standard">
                    <h3> Markov Process Models</h3>
                    <br>
                    <div class="textbox">
                    <p> 
                    A <i>Markov Process</i> is a random process in which the future state is not dependent on past states, but only on the present. (It is memory-less.) <br><br>

                    These types of models are used extensively in evolutionary modeling, and in many other fields. Made up of a set of discrete states, starting frequencies, and a mechanism for transitioning between states. <br><br>

                    Can be used to infer parameters (describing an unobserved process) that are most likely to produce observed data. For example, we model changes occurring in DNA sites as a Markov process.
                    </p></div>
                    <br>
                </div>
            </section>            


            <section class="future" data-transition="none">
                <div class="standard">
                    <h3> Markov Process Models</h3>
                    <br>
                    <div class="textbox">
                    <p style="text-align:center"> 
                    <a href=http://setosa.io/blog/2014/07/26/markov-chains/>http://setosa.io/blog/2014/07/26/markov-chains/</a>
                    </p></div>
                </div>
            </section>            





            <section class="future" data-transition="none">
                <div class="standard">
                    <h3> Markov Models for Evolutionary Inference</h3>
                    <br>
                    <div class="textbox">
                    <p> 
                    Markov models are widely used in evolution beyond tree inference. For example, to describe the evolution of a discrete character along branches of a tree. This is similar to the evolution of nucleotide substitutions, 
                    but can be the presence or absence of a binary trait, or a biogeographic state.
                    </p></div>
                    <img src=https://i2.wp.com/www.molecularecologist.com/wp-content/uploads/2015/09/Screen-Shot-2015-09-24-at-10.01.39-AM.png
                    style="width:35%">
                    <br>
                    <a href=https://www.molecularecologist.com/2015/09/measuring-dispersal-rate-in-neotropical-fishes-in-units-of-wallace/>
                    link to review of this paper.
                    </a>
                    <br>
                </div>
            </section>   









        <!-- SLIDES CLASS ENDING -->
        </div>

    <!-- REVEAL LINEAR CENTER CLASS ENDING -->
    </div>


    <!-- <div class="slide-number"></div> -->
    <!-- <div class="progress" style="display: block"><span style="width: 0px;"></span> -->
    <!-- JAVASCRIPT -->
    <script src="../../reveal/lib/js/head.min.js"></script>        
    <script src="../../reveal/js/reveal.js"></script>
    <script type="text/javascript" src="../../reveal/plugin/highlight/highlight.js"></script>
    <script>
        Reveal.initialize({
            controls:true, 
            progress:false, 
            loop:false, 
            history:true,
            center:false,
            fragmentInURL:true,
            margin:.05,
            minScale:0.1,
            maxScale:5,
            width:1600,
            height:900,
            theme:Reveal.getQueryHash().theme,
            transition:Reveal.getQueryHash().transition||"linear",
            slideNumber: true,
            // slideNumber: "h/v",

            dependencies: [
                //     { src: '../reveal/lib/js/classList.js', condition: function() { return !document.body.classList; } },
                { src: '../../reveal/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                { src: '../../reveal/plugin/zoom-js/zoom.js', async: !0, condition: function() {return!!document.body.classList }},
                { src: '../../reveal/plugin/math/math.js', async: true }
            ]
        });

        // Reveal.configure({ slideNumber: true })
        // Reveal.configure({ slideNumber: "c/t" })            

        // Reveal.initialize({controls:!1,progress:!0,history:!0,center:!0,margin:.05,minScale:.1,maxScale:5,width:960,height:720,theme:Reveal.getQueryHash().theme,transition:Reveal.getQueryHash().transition||"linear",math:{mathjax:"http://cdn.mathjax.org/mathjax/latest/MathJax.js",config:"TeX-AMS_HTML-full"},dependencies:[{src:"../reveal/lib/js/classList.js",condition:function(){return!document.body.classList}},{src:"../reveal/plugin/markdown/marked.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/markdown/markdown.js",condition:function(){return!!document.querySelector("[data-markdown]")}},{src:"../reveal/plugin/highlight/highlight.js",async:!0,callback:function(){hljs.initHighlightingOnLoad()}},{src:"../reveal/plugin/zoom-js/zoom.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/notes/notes.js",async:!0,condition:function(){return!!document.body.classList}},{src:"../reveal/plugin/math/math.js",async:!0}]});
    </script>
</body>
</html>